src/ReactSVG.tsx

```diff
@@ -95,12 +97,20 @@ export class ReactSVG extends React.Component<Props, State> {
 
       this.nonReactWrapper = this.reactWrapper.appendChild(nonReactWrapper)
 
+      const onError = (error: unknown) => {
+        this.removeSVG()
+        handleError(error)
+        this.setState({ hasError: true, isLoading: false })
+      }
       const afterEach = (error: Error | null, svg?: SVGSVGElement) => {
         if (error) {
-          this.removeSVG()
-
           if (!this._isMounted) {
-            afterInjection(error)
+            onError(error)
```

Comment:
Don't think we can get away with this as it's going to result in `setState` being called (L103) when the component is not mounted, which will throw an error. Will tweak things a bit.